toucan_connectors/odbc/odbc_connector.py

```diff
@@ -0,0 +1,44 @@
+import pandas as pd
+import pyodbc
+from pydantic import Field, constr
+
+from toucan_connectors.common import convert_to_qmark_paramstyle
+from toucan_connectors.toucan_connector import ToucanConnector, ToucanDataSource
+
+
+class OdbcDataSource(ToucanDataSource):
+    query: constr(min_length=1)
+
+
+class OdbcConnector(ToucanConnector):
+    """
+    Import data from ODBC apis
+    """
+
+    data_source_model: OdbcDataSource
+
+    connection_string: str = Field(..., description='The connection string')
+    autocommit: bool = False
```

Comment:
As this connectors are only to read data, should we expose this parameter?
We could set it to `False` in the `connection_params` and not allow users to change it.